CRM/ManualDirectDebit/Hook/PostProcess/Contribution/DirectDebitMandate.php

```diff
@@ -54,25 +78,40 @@ public function checkPaymentOptionToCreateMandate() {
     }
   }
 
+  /**
+   *  Launches all required processes after saving mandate
+   */
+  public function run() {
+    $this->setCurrentContactId($this->form->getVar('_entityId'));
+    $this->setCurrentMandateId();
+
+    $this->checkChangingMandateContribution();
+    $this->checkMailNotification();
+    $this->createMandateActivity();
+  }
+
   /**
    * Changes mandate for recurring contribution
    *
    * @return bool
    */
-  public function changeMandateForRecurringContribution() {
+  public function checkChangingMandateContribution() {
+    if (!isset($this->form->getVar('_submitValues')['recurrId']) || empty($this->form->getVar('_submitValues')['recurrId'])) {
```

Comment:
Same here, we can extract obtaining the value and assign it to a variable to improve readability.

Eg.
```php
$recurrID = $this->form->getVar('_submitValues')['recurrId'];

if (!isset($recuurrId) || empty($recurrId)) {
  ...
}
```